src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -851,13 +852,24 @@ private void recordSelectionForCastVoteRecord(
           selectedCandidate,
           cvr.getFractionalTransferValue());
       if (config.isTabulateByPrecinctEnabled()) {
-        precinctTallyTransfers
-            .get(cvr.getPrecinct())
-            .addTransfer(
-                currentRound,
-                cvr.getCurrentRecipientOfVote(),
-                selectedCandidate,
-                cvr.getFractionalTransferValue());
+        if (precinctIds.size() == 0) {
+          Logger.severe("\"Tabulate by Precinct\" enabled, but CVRs don't list precincts.");
+          throw new TabulationAbortedException(false);
+        }
```

Comment:
I don't see any reason not to have the constructor throw a `TabulationAbortedException` (in fact, it's already caught in the place where the constructor is called).

I'd personally move it there because I like cleaner logs and not going forward with other calcs to no end, but it's not a super big deal. Will go ahead and approve and let you make the call either way. 👍